### PR TITLE
Use Kyori/events for plugin events

### PR DESCRIPTION
**Pull request by Draycia**
One small step towards being platform agnostic.

Untested, needs heavy testing to ensure everything functions correctly.

### CODE REVIEW

File: src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java [hunk 1]
```diff
@@ -31,30 +35,38 @@ public TownyContext(@NonNull final CarbonChat carbonChat) {
 
       return true;
     });
-  }
 
-  @EventHandler(ignoreCancelled = true)
-  public void onChannelSwitch(final ChannelSwitchEvent event) {
-    final Object town = event.channel().context(KEY);
+    final TownyContext instance = this;
```
Comment (Proximyst) [inline src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java:39]: Why?

Comment (Draycia) [reply src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java:39]: Checkstyle - it complains about the omission of `this` which isn't possible due to the anonymous class.
After some trial and error I found this to be the easiest way to satisfy it.

Comment (Sauilitired) [reply src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java:39]: You can use TownyContext.this.<whatever> instead

Comment (Draycia) [reply src/main/java/net/draycia/carbon/channels/contexts/impl/TownyContext.java]: 4ad0434

**Pull request merged**